service/lib/agama/software/manager.rb

```diff
@@ -589,6 +598,30 @@ def installation_device(label)
         end
       end
 
+      # build URL for the Full installation repository
+      # @param path [String] Local path where the Full repository is mounted
+      # @param url_path [String] Path part of the resulting URL
+      # @return [String,nil] URL or `nil` if the Full repository device was not found
+      def full_repo_url(path, url_path)
+        # find the device which is mounted at the repository location
+        live_device = `findmnt -n -o SOURCE --target #{Shellwords.escape(path)}`.chomp
+        logger.info "Installation device: #{live_device}"
+        return nil unless live_device
+
+        # distinguish between DVD and hard disks/USB flash
```

Comment:
I guess this regexp is more than enough. I checked whether `findmnt` offered the same options than `lsblk` (e.g., `TRAN`) but it does not.